src/relay/pass/quantize.cc

```diff
@@ -413,14 +396,21 @@ Array<Expr> UnifyDTypeScale(const Array<Expr>& ref_args,
   // unify the data type
   CHECK_EQ(ref_args.size(), args.size());
   DataType dtype;
-  if (nptrs[0]->dtype == cfg->dtype_activation) {
-    DataType dtype = cfg->dtype_activation;
-    ret.Set(1, Cast(ret[1], dtype));
-  } else if (nptrs[1]->dtype == cfg->dtype_input) {
-    DataType dtype = cfg->dtype_input;
-    ret.Set(0, Cast(ret[0], dtype));
+  if (ret.size() == 2 && nptrs[1]->dtype == cfg->dtype_input) {
+    dtype = cfg->dtype_input;
   } else {
-    LOG(FATAL) << "should not touch here.";
+    dtype = cfg->dtype_activation;
+  }
+  for (size_t i = 0; i < ret.size(); ++i) {
```

Comment:
Hey @vinx13 , I guess we can remove this part with `rewrite_for_vta`, which will insert `cast` and `stop_fusion` in the end of residual block

Reply:
This part is needed when `rewrite_for_vta` is not enabled (`store_lowbit_output=False`), or we can remove this part in `rewrite_for_vta`?

Reply:
@vinx13  Current `Realize` pass has been used to achieve multiple goals: insert `stop_fusion`; decide datatype; unify scales for some operator, etc. It's better to decouple things for passes, I would also like to make `rewrite_for_vta` and `store_lowbit_output` enabled by default.